sql/hive-thriftserver/src/main/scala/org/apache/spark/sql/hive/thriftserver/FetchIterator.scala

```diff
@@ -0,0 +1,92 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.hive.thriftserver
+
+private[hive] sealed trait FetchIterator[A] extends Iterator[A] {
+  def fetchNext(): Unit
+
+  def setRelativePosition(diff: Long): Unit
+
+  def setAbsolutePosition(pos: Long): Unit
```

Comment:
Thanks,
Sorry, during the previous review I didn't remember the FETCH_PRIOR well that it's relative to the start of the previous block, not end of the current block. With that I changed my mind and think it would be more consistantly named and documented as:
```
// Begin a fetch block, forward from the current position
// Resets the fetch start offset
fetchNext()

// Begin a fetch block, moving the iterator back by offset from the start of the previous fetch block start.
// Resets the fetch start offset
fetchPrior(offset: Long)

// Begin a fetch block, moving the iterator to the given offset.
// Resets the fetch start offset
fetchAbsolute(pos: Long)
```

with setRelativePosition it's a bit confusing if it's relative to the current position, or the previous start offset.

Reply:
Thanks,
I think it becomes much better than previous versions.
I applied patch and rewrite test code of FetchIterator.
Do I create issue & PR in Apache Spark repo?